DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs

```diff
@@ -1534,8 +1544,8 @@ private void MapSharer_HandleMapDownloadFailed(SHA1EventArgs e)
 
         private void MapSharer_HandleMapDownloadComplete(SHA1EventArgs e)
         {
-            Logger.Log("Map " + e.SHA1 + " downloaded, parsing.");
-            string mapPath = "Maps/Custom/" + e.SHA1;
+            Logger.Log("Map " + e.MapName + "_" + e.SHA1 + " downloaded, parsing.");
+            string mapPath = "Maps/Custom/" + e.MapName + "_" + e.SHA1;
```

Comment:
two here for use of your new utility function

Reply:
done